engine.php

```diff
@@ -197,7 +197,16 @@ private function parse_master_response($data, &$index, &$count, &$servers)
             {
                 $ip = [];
                 foreach ( range(0, 3) as $i )
+                {
                     $ip[] = (string)ord($nextbyte());
+                    // ipv4 address field starts with '\'
+                    // but serverResponse packet also ends with '\'
+                    // if end of packet is reached, do not parse ipv4
+                    if ( feof($buffer) )
+                        break;
+                }
+                if ( feof($buffer) )
+                    break;
                 $port = ord($nextbyte()) << 8;
                 $port |= ord($nextbyte());
                 $ip = implode(".", $ip);
```

Comment:
I wonder why `if ( $ip !== "0.0.0.0" )` doesn't prevent the bogus server from appearing? But it doesn't matter, checking for eof is still a good idea regardless.

Reply:
We may remove this test as it looks to have been an ugly workaround.

In any way I wonder why php7 introduced a regression in that code.

Reply:
Hu, this code is not present on code running on our server, maybe this workaround is older than the code we run.
Well, it's better to have a proper code than to workaround the garbage produced by the wrong code.
I'll remove this workaround from the repository then pull the changes on the server.